src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1430,8 +1430,10 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
         continue;
       }
       failureMessage = canPassThroughCanal(canalAttachment, units, player, data);
-      if (!failureMessage.isPresent()) {
```

Comment:
Umm. Pretty sure that won't work... 

That doesn't even take into consideration the new property which is the whole point of the change.